PWGJE/EMCALJetTasks/AliAnalysisTaskRhoSparse.cxx

```diff
@@ -131,57 +134,91 @@ Bool_t AliAnalysisTaskRhoSparse::Run()
   static Double_t rhovec[999];
   Int_t NjetAcc = 0;
   Double_t TotaljetAreaPhys=0;
+  Double_t TotalAreaCovered=0;
   Double_t TotalTPCArea=2*TMath::Pi()*0.9;
 
   // push all jets within selected acceptance into stack
   for (Int_t iJets = 0; iJets < Njets; ++iJets) {
 
-    // exlcuding leading background jets (could be signal)
-    if (iJets == maxJetIds[0] || iJets == maxJetIds[1])
-      continue;
-
     AliEmcalJet *jet = static_cast<AliEmcalJet*>(fJets->At(iJets));
     if (!jet) {
       AliError(Form("%s: Could not receive jet %d", GetName(), iJets));
       continue;
     } 
 
+    // Take into account the area of real jets (no pure ghost jets)
+    // and also the area of later excluded jets.
+    // Some of these jets do not contribute to the rho calculation
+    // but to the factor with which this rho is scaled down
+    if (fExcludeAreaExcludedJets==0 && jet->GetNumberOfTracks()>0)
+    {
+      TotaljetAreaPhys+=jet->Area();
+      // Total area of all found jets ghost+phyical jets. This is a proxy
+      // for the available detector area in which the rho could have been calculated
+      TotalAreaCovered+=jet->Area();
+    }
+    // Exclude leading background jets (could be signal)
+    if (iJets == maxJetIds[0] || iJets == maxJetIds[1])
```

Comment:
Can these lines 161 + 162 be moved before the line 149?

Reply:
Hi @fkrizek,

please look at L191. If you set fExcludeAreaExcludedJets==1 which I recommended, then the area is summed after the jets are excluded and L153-159 is ignored. I think this is what you want

Reply:
Hi @efranzis.

yes, you are right!   I guess you can commit the code then.
Cheers
Filip